lib/Mojolicious/Guides/Testing.pod

```diff
@@ -0,0 +1,802 @@
+application output):
+
+  $ cd my_app
+  $ mkdir log
+  $ prove -lv t
+  t/basic.t ..
+  ok 1 - GET /
+  ok 2 - 200 OK
+  ok 3 - content is similar
+  1..3
+  ok
+  All tests successful.
+  Files=1, Tests=3,  0 wallclock secs ( 0.03 usr  0.01 sys +  0.33 cusr  0.07 csys =  0.44 CPU)
+  Result: PASS
+
+The boilerplate test file looks like this:
+
+  use Mojo::Base -strict;
+
+  use Test::More;
+  use Test::Mojo;
+
+  my $t = Test::Mojo->new('MyApp');
+  $t->get_ok('/')->status_is(200)->content_like(qr/Mojolicious/i);
+
+  done_testing();
+
+Here we can see our application class name C<MyApp> is passed to the
+L<Test::Mojo> constructor. Under the hood, L<Test::Mojo> creates a new
+L<Mojo::Server> instance, loads C<MyApp> (which we just created), and runs the
+application. We write our tests with relative URLs because L<Test::Mojo> takes
+care of getting the request to the running test application (since its port
+may change between runs).
+
+=head3 Testing with configuration data
+
+We can alter the behavior of our application using environment variables (such
+as C<MOJO_MODE>) and through configuration values. One nice feature of
+L<Test::Mojo> is its ability to pass configuration values directly from its
+constructor.
+
+Let's modify our application and add a "feature flag" to enable a new feature
+when the C<enable_weather> configuration value is set:
+
+  # Load configuration from hash returned by "my_app.conf"
+  my $config = $self->plugin('Config');
+
+  # Normal route to controller
+  $r->get('/')->to('example#welcome');
+
+  # NEW: this route only exists when 'enable_weather' is set in the configuration
+  if ($config->{enable_weather}) {
+      $r->get('/weather')->to(cb => sub { shift->render(text => "It's hot! 🔥") }
+  }
+
+To test this new feature, we don't even need to create a configuration file—we
+can simply pass the configuration data to the application directly via
+L<Test::Mojo>'s constructor:
+
+  my $t = Test::Mojo->new('MyApp', { enable_weather => 1 });
+  $t->get_ok('/')->status_is(200)->content_like(qr/Mojolicious/i);
+  $t->get_ok('/weather')->status_is(200)->content_like(qr/🔥/);
+
+When we run these tests, L<Test::Mojo> will pass this configuration data to
+the application, which will cause it to create a special C</weather> route
+that we can access in our tests. Unless C<enable_weather> is set in a
+configuration file, this route will not exist when the application
+runs. Feature flags like this allow us to do soft rollouts of features,
+targeting a small audience for a period of time. Once the feature has been
+proven, we can refactor the conditional and make it a full release.
+
+This example shows how easy it is to start testing a Mojolicious application
+and how to set specific application configuration directives from a test file.
+
+=head3 Testing application helpers
+
+Let's say we register a helper in our application to generate an HTTP Basic
+Auth header value:
```

Comment:
Ah, I misunderstood. This is fixed (and I still like the full header value as a hashref).